src/libraries/System.Formats.Tar/tests/TarWriter/TarWriter.WriteEntry.Entry.Roundtrip.Tests.cs

```diff
@@ -245,5 +249,71 @@ public void PaxExtendedAttributes_DoNotOverwritePublicProperties_WhenLargerThanL
             Assert.Equal(writeEntry.ModificationTime, readEntry.ModificationTime);
             Assert.Equal(writeEntry.LinkName, readEntry.LinkName);
         }
+
+        [Theory]
+        [InlineData(TarEntryFormat.V7, LegacyMaxFileSize)]
+        [InlineData(TarEntryFormat.Ustar, LegacyMaxFileSize)]
+        [InlineData(TarEntryFormat.Gnu, LegacyMaxFileSize)]
+        [InlineData(TarEntryFormat.Pax, LegacyMaxFileSize)]
+        // Pax supports unlimited size files.
+        [InlineData(TarEntryFormat.Pax, LegacyMaxFileSize + 1)]
+        public void WriteEntry_LongFileSize(TarEntryFormat entryFormat, long size)
```

Comment:
The new tests create ~8 Gb files, locally it took me 153s to run all tests now, when previously was taken ~5 secs. Any suggestions if this can be improved? 

Of course, I will add `[OuterLoop]`.

Reply:
And I think there is also another attribute that ensures the test does not run in parallel with others, which would help avoid running out of memory.

Reply:
> avoid running out of ~~memory~~.

disk.

Reply:
I meant memory. I don't think we should be using disk at all.

Reply:
If memory then disable for 32 bit?

Reply:
We can't use MemoryStream, it uses an `int _position` and throws for these cases.
```
System.Formats.Tar.Tests.TarWriter_WriteEntry_Roundtrip_Tests.WriteEntry_LongFileSize(entryFormat: V7, size: 8589934591) [FAIL]
        System.IO.IOException : Stream was too long.
        Stack Trace:
          C:\repos\runtime\src\libraries\System.Private.CoreLib\src\System\IO\MemoryStream.cs(600,0): at System.IO.MemoryStream.Write(Byte[] buffer, Int32 offset, Int32 count)
          C:\repos\runtime\src\libraries\System.Private.CoreLib\src\System\IO\Stream.cs(79,0): at System.IO.Stream.CopyTo(Stream destination, Int32 bufferSize)
          C:\repos\runtime\src\libraries\System.Private.CoreLib\src\System\IO\Strategies\BufferedFileStreamStrategy.cs(915,0): at System.IO.Strategies.BufferedFileStreamStrategy.CopyTo(Stream destination, Int32 bufferSize)
          C:\repos\runtime\src\libraries\System.Private.CoreLib\src\System\IO\FileStream.cs(521,0): at System.IO.FileStream.CopyTo(Stream destination, Int32 bufferSize)
          C:\repos\runtime\src\libraries\System.Private.CoreLib\src\System\IO\Stream.cs(51,0): at System.IO.Stream.CopyTo(Stream destination)
          C:\repos\runtime\src\libraries\System.Formats.Tar\src\System\Formats\Tar\TarHeader.Write.cs(611,0): at System.Formats.Tar.TarHeader.WriteData(Stream archiveStream, Stream dataStream, Int64 actualLength)
```

Reply:
Oh, of course. MemoryStream is limited to a single byte array. You'd need to create a wrapper stream around several MemoryStreams, or just a fake stream that returned whatever you wanted for a particular offset.

Generally speaking, what's the level of support in Streams for very large stores? It's funny that there's a Seek() taking a long, and Position is a long, but Read only take an int offset. Do we have an issue tracking that?

Reply:
I do think in memory is the way to go.

@MattGal remind us, in Helix, what's the minimum $TEMP disk space you can be sure is available? As I seem to recall that on some queues, it can be only a few GB.

Reply:
Helix machines will keep taking work as long as the drive their "work" directory is on has > 3172 MB of free space.  Looking at this PR it sounds like you want to create 8+ GB files so there's no guarantee this test can work there; perhaps if you must do this you could check the free space first and nerf the test out.  

Folks haven't directly linked the ones they're looking at, [so grabbing one from RedHat 7 ](https://helixre107v0xdeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-pull-76707-merge-bd7abad596984c68a5/System.Formats.Tar.Tests/1/console.efd2eebf.log?helixlogtype=result) it's hitting this problem even in the case of running on machines that start with ~48 GB free space.

My guess here is that there are multiple tests running in parallel, that create > 8 GB files simultaneously and eat the entire disk quickly, given that the link above ran on a machine that definitely had ~46 GB free when the work item started.

Reply:
Yes, that's another thing to do if you're making huge files -- run them alone, serialized and not in parallel. We have prior art for using Xunit collections to do this in the runtime repo for certain tests.

Reply:
Most Helix test VMs are of the Standard D2* variety (various intel versions and AMD EPYC used in different places for reasons). These machines have 8 GB total space, so it's _probably_ a no, but more likely a "varies by OS" problem, since virtualization should allow this in some environments.